crates/bevy_pbr/src/render_graph/pipeline/pbr.frag

```diff
@@ -0,0 +1,322 @@
+
+void main() {
+    vec4 output_color = base_color;
+#ifdef STANDARDMATERIAL_BASE_COLOR_TEXTURE
+    output_color *= texture(sampler2D(StandardMaterial_base_color_texture,
+                                      StandardMaterial_base_color_texture_sampler),
+                            v_Uv);
+#endif
+
+#ifndef STANDARDMATERIAL_UNLIT
+    // calculate non-linear roughness from linear perceptualRoughness
+    float roughness = perceptualRoughnessToRoughness(perceptual_roughness);
+
+    vec3 N = normalize(v_WorldNormal);
+
+    vec3 V = normalize(CameraPos.xyz - v_WorldPosition.xyz);
+    // Neubelt and Pettineo 2013, "Crafting a Next-gen Material Pipeline for The Order: 1886"
+    float NdotV = max(dot(N, V), 1e-4);
+
+    // Remapping [0,1] reflectance to F0
+    // See https://google.github.io/filament/Filament.html#materialsystem/parameterization/remapping
+    vec3 F0 = 0.16 * reflectance * reflectance * (1.0 - metallic) + output_color.rgb * metallic;
+
+    // Diffuse strength inversely related to metallicity
+    vec3 diffuseColor = output_color.rgb * (1.0 - metallic);
+
+    // accumulate color
+    vec3 light_accum = vec3(0.0);
+    for (int i = 0; i < int(NumLights.x) && i < MAX_LIGHTS; ++i) {
+        Light light = SceneLights[i];
+
+        vec3 lightDir = light.pos.xyz - v_WorldPosition.xyz;
+        vec3 L = normalize(lightDir);
+
+        float rangeAttenuation =
+            getDistanceAttenuation(lightDir, light.inverseRadiusSquared);
+
+        vec3 H = normalize(L + V);
+        float NoL = saturate(dot(N, L));
+        float NoH = saturate(dot(N, H));
+        float LoH = saturate(dot(L, H));
+
+        vec3 specular = specular(F0, roughness, H, NdotV, NoL, NoH, LoH);
+        vec3 diffuse = diffuseColor * Fd_Burley(roughness, NdotV, NoL, LoH);
+
+        // Lout = f(v,l) Φ / { 4 π d^2 }⟨n⋅l⟩
+        // where
+        // f(v,l) = (f_d(v,l) + f_r(v,l)) * light_color
+        // Φ is light intensity
+
+        // our rangeAttentuation = 1 / d^2 multiplied with an attenuation factor for smoothing at the edge of the non-physical maximum light radius
+        // It's not 100% clear where the 1/4π goes in the derivation, but we follow the filament shader and leave it out
+
+        // See https://google.github.io/filament/Filament.html#mjx-eqn-pointLightLuminanceEquation
+        // TODO compensate for energy loss https://google.github.io/filament/Filament.html#materialsystem/improvingthebrdfs/energylossinspecularreflectance
+        // light.color.rgb is premultiplied with light.intensity on the CPU
+        light_accum +=
+            ((diffuse + specular) * light.color.rgb) * (rangeAttenuation * NoL);
+    }
+
+    vec3 diffuse_ambient = EnvBRDFApprox(diffuseColor, 1.0, NdotV);
+    vec3 specular_ambient = EnvBRDFApprox(F0, perceptual_roughness, NdotV);
+
+    output_color.rgb = light_accum + (diffuse_ambient + specular_ambient) * AmbientColor;
+
+    // tone_mapping
+    output_color.rgb = reinhard_luminance(output_color.rgb);
```

Comment:
I think technically we're calculating luma here as output_color is sRGB?

Reply:
I guess it doesn't really matter. A "luminance function" becomes a "luma function" if you pass in an sRGB value.

Reply:
Oh, fair point. Is it worth keeping this in a separate vec4 and only writing to `output_color` once at the end? It saves a bunch of gamma corrections, I suspect?

Reply:
sRGB is already "gamma corrected" by nature, so theres no extra work being done here. Its "just" a vec4. I think we can just leave this as-is.